cmd/gardener-node-agent/app/app.go

```diff
@@ -205,19 +260,46 @@ func run(ctx context.Context, cancel context.CancelFunc, log logr.Logger, cfg *c
 	}
 
 	log.Info("Creating directory for temporary files", "path", nodeagentv1alpha1.TempDir)
-	fs := afero.Afero{Fs: afero.NewOsFs()}
 	if err := fs.MkdirAll(nodeagentv1alpha1.TempDir, os.ModeDir); err != nil {
 		return fmt.Errorf("unable to create directory for temporary files %q: %w", nodeagentv1alpha1.TempDir, err)
 	}
 
+	bootstrapRunnables := []manager.Runnable{
+		&bootstrappers.KubeletBootstrapKubeconfig{Log: log.WithName("kubelet-bootstrap-kubeconfig-creator"), FS: fs, APIServerConfig: cfg.APIServer},
+	}
+
+	var machineName string
+	if features.DefaultFeatureGate.Enabled(features.NodeAgentAuthorizer) {
+		machineName, err = fetchMachineName(fs)
+		// TODO(oliver-goetz): fetching the machine name from node label is migration code remove this migration code when feature gate NodeAgentAuthorizer is removed.
+		// Fetching the machine name from node label is migration for existing nodes when the feature gate is activated, because there is no file with the machine name yet.
```

Comment:
```suggestion
		// Fetching the machine name from node label is for migrating existing nodes when the feature gate is activated, because there is no file with the machine name yet.
```